### PR TITLE
fix(ivy): unable to override ComponentFactoryResolver provider in tests

### PR DESCRIPTION
**Pull request by atscott**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Cannot override ComponentFactoryResolver in TestBed tests. The test added in this PR was failing in Ivy without the change in r3_injector

### CODE REVIEW

File: packages/core/src/di/r3_injector.ts [hunk 1]
```diff
@@ -119,12 +119,12 @@ export class R3Injector {
     // Start off by creating Records for every provider declared in every InjectorType
     // included transitively in `def`.
```
Comment (AndrewKushnir) [inline packages/core/src/di/r3_injector.ts]: It looks like this comment should also be updated to describe the updated sequence. I'd also propose to include additional information (as a comment) on why keeping this sequence is important (i.e. give priority to providers declared in user's code).

File: packages/platform-browser/test/testing_public_spec.ts [hunk 2]
```diff
@@ -1042,3 +1052,13 @@ Did you run and wait for 'resolveComponentResources()'?` :
...
   });
 }
+describe('override', () => {
+
+  it('defined in core - injectable providedIn', () => {
+    const sanitizerMock = jasmine.createSpyObj('sanitizer', ['sanitize']);
+    TestBed.overrideProvider(Sanitizer, {useValue: sanitizerMock});
+    expect(TestBed.get(Sanitizer)).toEqual(sanitizerMock);
+  });
+
```
Comment (AndrewKushnir) [inline packages/platform-browser/test/testing_public_spec.ts]: Extra new lines.

File: packages/core/src/di/r3_injector.ts (hunk 1 above)
Comment (atscott) [reply packages/core/src/di/r3_injector.ts]: Done.

File: packages/platform-browser/test/testing_public_spec.ts (hunk 2 above)
Comment (atscott) [reply packages/platform-browser/test/testing_public_spec.ts]: Done. Actually meant to delete that completely.

**Comment by AndrewKushnir** [comment]
Thanks for making additional changes @atscott, the changes LGTM 👍 

@kara could you please have a look at these changes as well?

Thank you.

**Review by kara — APPROVED** [review]
LGTM

**Comment by matsko** [comment]
Presubmit = http://test/OCL:268265870:BASE:268265890:1568139116910:11bb378d

Ivy Presubmit = http://test/OCL:268265870:BASE:268266104:1568139169481:22b0f257

**Pull request closed without merging**